coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/parsetree/ParseTreeBuilder.java

```diff
@@ -222,38 +222,55 @@ private SqlNode visitLateralViewInternal(ASTNode node, ParseContext ctx, boolean
 
   private SqlNode visitLateralViewExplode(List<SqlNode> sqlNodes, List<SqlNode> aliasOperands,
       SqlCall tableFunctionCall, boolean isOuter) {
-    checkState(aliasOperands.size() == 3);
+    final int operandCount = aliasOperands.size();
+    // array if operandCount == 3, map if operandCount == 4
+    checkState(operandCount == 3 || operandCount == 4,
+        format("Unsupported LATERAL VIEW EXPLODE operand number: %d", operandCount));
     // TODO The code below assumes LATERAL VIEW is used with UNNEST EXPLODE only. It should be made more generic.
```

Comment:
do you know if there is an issue created for this? If not, do you mind creating one and reference it here? (and please create a parallel internal BDP jira in the backlog)

Reply:
Opened an issue (https://github.com/linkedin/coral/issues/108) and created a ticket.